### PR TITLE
[Components with more than 1 service] LPS-190421 Refactor ServerStatistics classes from portal-monitoring module

### PR DESCRIPTION
**Pull request by RenanUmV**
[LPS-190421](https://liferay.atlassian.net/browse/LPS-190421)

### CODE REVIEW

**Review by rafaprax — APPROVED** [review]

**Comment by Kvale1733** [comment]
@RenanUmV

**Comment by dantewang** [comment]
@RenanUmV @rafaprax Could you double check that after these changes, should these renamed classes remain in the current packages? It seems to me that a more clear way is to keep the `*Statistics` classes, extract the processes into another package.

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:startup

File: modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/processor/PortalRequestDataSampleProcessor.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.monitoring.internal.processor;
+
```
Comment (rafaprax) [inline modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/processor/PortalRequestDataSampleProcessor.java:7]: HEy @RenanUmV, please rename the package to `com.liferay.portal.monitoring.internal.data.sample.processor` instead. Thx

File: modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/statistics/portal/ServerStatistics.java [hunk 2]
```diff
@@ -24,12 +24,12 @@
...
-@Component(
-	enabled = false, property = "namespace=com.liferay.monitoring.Portal",
-	service = {DataSampleProcessor.class, ServerStatistics.class}
-)
-public class ServerStatistics
-	implements DataSampleProcessor<PortalRequestDataSample> {
+@Component(service = ServerStatistics.class)
+public class ServerStatistics {
+
+	public Company getCompany(long companyId) throws PortalException {
```
Comment (rafaprax) [inline modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/statistics/portal/ServerStatistics.java:30]: Hey @RenanUmV, you do not need to put this method here. Just inline it. Thx

File: modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/statistics/portal/ServerStatistics.java [hunk 3]
```diff
@@ -24,12 +24,12 @@
...
  */
-@Component(
-	enabled = false, property = "namespace=com.liferay.monitoring.Portal",
-	service = {DataSampleProcessor.class, ServerStatistics.class}
-)
-public class ServerStatistics
-	implements DataSampleProcessor<PortalRequestDataSample> {
+@Component(service = ServerStatistics.class)
+public class ServerStatistics {
+
```
Comment (rafaprax) [inline modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/statistics/portal/ServerStatistics.java:29]: Hey @RenanUmV, now this class looks like a helper class. Could you please rename this to reflect this change?

Thx

File: modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/statistics/portal/ServerStatistics.java [hunk 4]
```diff
@@ -24,12 +24,12 @@
...
  * @author Brian Wing Shun Chan
  */
-@Component(
-	enabled = false, property = "namespace=com.liferay.monitoring.Portal",
-	service = {DataSampleProcessor.class, ServerStatistics.class}
-)
-public class ServerStatistics
-	implements DataSampleProcessor<PortalRequestDataSample> {
+@Component(service = ServerStatistics.class)
+public class ServerStatistics {
```
Comment (rafaprax) [inline modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/statistics/portal/ServerStatistics.java:28]: Hey @RenanUmV, this service should be enabled by default? Please check that. Thx

File: modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/processor/ServiceRequestDataSampleProcessor.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+			_serviceStatistics.put(className, serviceStatistics);
+		}
+
+		serviceStatistics.processDataSample(serviceRequestDataSample);
+	}
+
+	private final Map<String, ServiceStatistics> _serviceStatistics =
+		new ConcurrentHashMap<>();
+
```
Comment (rafaprax) [inline modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/processor/ServiceRequestDataSampleProcessor.java:50]: Hey @RenanUmV, this seems wrong. We have the same map in the other class. Could you please check that? Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Review by rafaprax — CHANGES_REQUESTED** [review]
Hey @RenanUmV, add the enabled property to the created/extracted classes and break each commit into two ones (one for the extraction and another to the rename)

**Comment by rafaprax** [comment]
Hey @RenanUmV, move the extracted DataSampleProcessors into the proper packages(portal, portlet and service). Please fix and resend.

**Pull request closed without merging**